validation/src/main/java/io/micronaut/validation/executable/ExecutableVisitor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.validation.executable;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.inject.ast.MethodElement;
+import io.micronaut.inject.ast.ParameterElement;
+import io.micronaut.inject.visitor.TypeElementVisitor;
+import io.micronaut.inject.visitor.VisitorContext;
+
+/**
+ * A {@link io.micronaut.inject.visitor.TypeElementVisitor} that validates primitive types can't have a Nullable annotation.
+ *
+ * @author Iván López
+ * @since 2.0.1
+ */
+@Internal
+public class ExecutableVisitor implements TypeElementVisitor<Object, Object> {
+
+    @NonNull
+    @Override
+    public VisitorKind getVisitorKind() {
+        return VisitorKind.ISOLATING;
+    }
+
+    @Override
+    public void visitMethod(MethodElement element, VisitorContext context) {
+        ParameterElement[] parameters = element.getParameters();
```

Comment:
We need to ensure the `MethodElement` is annotated with `@Executable`. Can you look into why `TypeElementVisitor<Object, Executable>` is not working?

Reply:
If we can limit this as much as possible to a pair of meta annotations it is better for incremental compilation

Reply:
@jameskleeh fixed now. See my comment: https://github.com/micronaut-projects/micronaut-core/pull/3698#discussion_r452714413 and https://github.com/micronaut-projects/micronaut-core/pull/3698/commits/5197f8a986e7d5bb34fd73ce5e679ee4fb8465d5.

@graemerocher Can you please explain what do you mean? I've pushed a commit that makes it work with `TypeElementVisitor<Object, Executable>`. Do you mean making it for something that it's not `Object`? Like  `TypeElementVisitor<Any-other-thing-here, Executable>?

Reply:
If you have been able to limit it to just `@Executable` that is probably ok. In general `<Object, Object>` is bad for incremental compilation since it has to visit every annotation which breaks the way Gradle's incremental compilation works. Currently this is only a problem with `GraalTypeElementVisitor` I believe, which is the only one that visits everything (really we should try and change that visitor or restructure it to visit fewer annotations)

Reply:
Thanks for the clarification.

Regarding the `GraalTypeElementVisitor`, if you have any idea or guidance in mind, please open an issue explaining what you have in mind and I can give it a try.